applications/tasks/src/kz_task_worker.erl

```diff
@@ -107,7 +107,9 @@ loop(IterValue, State=#state{task_id = TaskId
                             ,extra_args = ExtraArgs
                             }) ->
     case kz_csv:take_row(get(?IN)) of
-        'eof' -> teardown(State, API, IterValue);
+        'eof' ->
+            _ = is_task_successful(TaskId, API, ExtraArgs, FAssoc, 'eof', IterValue),
```

Comment:
Because `teardown/3`  did not call `Module:TaskName/3`.
Tha main idea is to make a call `Module:TaskName/3`. where 3rd argument will be 'eof'.